cl/opinion_page/views.py

```diff
@@ -948,6 +1098,159 @@ async def view_authorities(
     )
 
 
+async def check_flag_exists(flag_name: str) -> bool:
+    return await sync_to_async(
+        waffle.get_waffle_flag_model().objects.filter(name=flag_name).exists
+    )()
+
+
+@never_cache
+async def view_opinion(request: HttpRequest, pk: int, _: str) -> HttpResponse:
+    """View for displaying opinions."""
+
+    flag_exists = await check_flag_exists("ui_flag_for_o")
+    if flag_exists:
+        ui_flag_for_o = await sync_to_async(waffle.flag_is_active)(
+            request, "ui_flag_for_o"
+        )
+        user_flag_active = await sync_to_async(waffle.flag_is_active)(
+            request.user, "ui_flag_for_o"
+        )
+        if ui_flag_for_o or user_flag_active:
+            return await render_opinion_view(request, pk, "opinions")
+    return await view_opinion_old(request, pk, "str")
```

Comment:
We can simplify this view:

```suggestion
    ui_flag_for_o = await sync_to_async(waffle.flag_is_active)(
        request, "ui_flag_for_o"
    )
    if ui_flag_for_o:
        return await render_opinion_view(request, pk, "opinions")
    return await view_opinion_old(request, pk, "str")
```